cmd/regctl/digest.go

```diff
@@ -30,18 +46,74 @@ func NewDigestCmd(rootOpts *rootCmd) *cobra.Command {
 		RunE:   digestOpts.runDigest,
 	}
 
+	digestCmd.Flags().StringVarP(&digestOpts.decompress, "decompress", "", "", "Decompress the bytes, using one of 'auto', 'zstd', 'gzip'")
 	return digestCmd
 }
 
 func (digestOpts *digestCmd) runDigest(cmd *cobra.Command, args []string) error {
+	var reader io.Reader = bufio.NewReader(cmd.InOrStdin())
+
+	var decompress compression
+	var err error
+
+	// Detect the compression in stdin
+	if digestOpts.decompress == "auto" {
+		decompress, err = getCompression(reader.(*bufio.Reader))
+		if err != nil {
+			return err
+		}
+	} else if digestOpts.decompress == "gzip" {
+		decompress = compressionGzip
+	} else if digestOpts.decompress == "zstd" {
+		decompress = compressionZstd
+	}
+
+	if decompress == compressionGzip {
+		reader, err = gzip.NewReader(reader)
+		if err != nil {
+			return err
+		}
+	} else if decompress == compressionZstd {
+		reader = zstd.ZipDecompressor()(reader)
+	}
```

Comment:
With pkg/archive, this could be switched to:

```suggestion
	var reader io.Reader = cmd.InOrStdin()

	if digestOpts.decompress {
		reader, err = archive.Decompress(reader)
		if err != nil {
			return err
		}
	}
```

Reply:
Done.